indev/libinput.c

```diff
@@ -186,6 +185,7 @@ bool libinput_set_file_state(libinput_drv_state_t *state, char* dev_name)
  */
 void libinput_init(void)
 {
+  memset(&default_state, 0, sizeof(libinput_drv_state_t));
```

Comment:
Doesn't this undo `default_state = { .event_lock = PTHREAD_MUTEX_INITIALIZER, };`?

I'm also not seeing any call to `pthread_mutex_init` in the corresponding [unl0kr MR](https://gitlab.com/cherrypicker/unl0kr/-/merge_requests/21/diffs). Could it be that we're accidentally using `event_lock` from `default_state` in the dynamic states in unl0kr? They are cleared with `lv_memzero` so `event_lock` should always result in `0` which might be what `PTHREAD_MUTEX_INITIALIZER` returns in this case.

I think it would actually be good to call `pthread_mutex_init` inisde of `libinput_init_state` because the caller shouldn't have to know anything about the thread handling inside of the driver. You may just have to make sure that `libinput_init` doesn't initialize the lock in `default_state` twice by calling `libinput_init_state` but not sure if that would actually be a problem

Reply:
Proposal for how to fix this: https://github.com/calebccff/lv_drivers/pull/2